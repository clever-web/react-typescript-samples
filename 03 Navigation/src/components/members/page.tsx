import * as React from 'react';

export const MembersPage: React.StatelessComponent<{}> = () => {
  return (
    <div className="row">
      <h2> Members Page</h2>
    </div>
  );
}
