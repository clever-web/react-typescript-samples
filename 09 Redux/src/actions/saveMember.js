var memberAPI_1 = require('../api/memberAPI');
var memberFormValidator_1 = require('../validations/memberFormValidator');
var saveMember = function (member) {
    var errorsSave = memberFormValidator_1.default.validateMember(member);
    if (errorsSave.isEntityValid) {
        memberAPI_1.default.saveAuthor(member);
    }
    return {
        type: 'MEMBER_SAVE',
        errors: errorsSave
    };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = saveMember;
//# sourceMappingURL=saveMember.js.map