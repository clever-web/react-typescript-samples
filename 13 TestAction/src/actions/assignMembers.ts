
const assignMembers = (members : any) => {
   return {
     type: 'MEMBERS_ASSIGN'
     ,members: members
   }
 }

export default assignMembers;
