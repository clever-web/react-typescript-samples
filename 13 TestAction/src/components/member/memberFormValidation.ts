import {
  Validators, ValidationConstraints, createFormValidation
} from 'lc-form-validation';

const validationConstraints: ValidationConstraints = {
  fields: {
    login: [
      { validator: Validators.required },
      {
        validator: Validators.minLength,
        customParams: { length: 3 },
      },
    ],
    avatar_url: [
      { validator: Validators.required },
    ]
  },
};

export const memberFormValidation = createFormValidation(validationConstraints);
