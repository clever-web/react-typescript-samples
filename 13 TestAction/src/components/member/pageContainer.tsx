import * as React from 'react';
import { connect } from 'react-redux';
import { State } from '../../reducers';
import { MemberEntity } from '../../model';
import { fetchMemberByIdAction } from './actions/fetchMemberById';
import { memberFieldChangeAction } from './actions/memberFieldChange';
import { saveMemberAction } from './actions/saveMember';
import { MemberPage } from './page';

const mapStateToProps = (state: State, ownProps: any) => (
  //if (ownProps.match===undefined) 
    {
      memberId: Number(ownProps && ownProps.match && ownProps.match.params && ownProps.match.params.id) || 0,
      member: state.member,
      memberErrors: state.memberErrors,
    }
  );

const mapDispatchToProps = (dispatch) => ({
  fetchMemberById: (id: number) => dispatch(fetchMemberByIdAction(id)),
  onChange: (member: MemberEntity, fieldName: string, value: string) =>
    dispatch(memberFieldChangeAction(member, fieldName, value)),
  onSave: (member: MemberEntity) => dispatch(saveMemberAction(member)),
});

const MemberPageContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(MemberPage);

export default MemberPageContainer;
