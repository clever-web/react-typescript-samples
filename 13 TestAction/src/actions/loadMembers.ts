import * as Q from 'q';
import MemberEntity from '../api/memberEntity';
import memberAPI from '../api/memberAPI';
import assignMembers from './assignMembers';

function loadMembers() {

  // Invert control!
  // Return a function that accepts `dispatch` so we can dispatch later.
  // Thunk middleware knows how to turn thunk async actions into actions.

  return dispatcher => {
    var promise : Q.Promise<MemberEntity[]>;

    promise = memberAPI.getAllMembersAsync();

    promise.then(
      data => dispatcher(assignMembers(data))
    );

    return promise;
  };
}

export default loadMembers;
