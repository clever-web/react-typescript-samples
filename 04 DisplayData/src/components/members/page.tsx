import * as React from 'react';
import { MemberEntity } from '../../model';
import { memberAPI } from '../../api/member';

interface State {
  members: MemberEntity[];
}

export class MembersPage extends React.Component<Props, State> {

  constructor() {
    super();
    this.state = { members: [] };
  }

  public componentWillMount() {
    this.state.members = memberAPI.fetchMembers();
  }

  public render() {
    var CreateMemberRow = function (member: MemberEntity) {
      return (
        <tr key={member.id}>
          <td>
            <img src={member.avatar_url} className="avatar" />
          </td>
          <td>
            <span>{member.id}</span>
          </td>
          <td>
            <span>{member.login}</span>
          </td>
        </tr>
      )
    }

    return (
      <div className="row">
        <h2> Members Page</h2>
        <table className="table">
          <thead>
            <tr>
              <th>
                Avatar
                </th>
              <th>
                Id
                </th>
              <th>
                Name
                </th>
            </tr>
          </thead>
          <tbody>
            {this.state.members.map(CreateMemberRow, this)}
          </tbody>
        </table>
      </div>
    );
  }
}
