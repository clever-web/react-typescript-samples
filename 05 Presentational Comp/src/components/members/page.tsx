import * as React from 'react';
import { MemberEntity } from '../../model';
import { memberAPI } from '../../api/member';
import { MemberHeader } from './memberHeader';
import { MemberRow } from './memberRow';

interface State {
  members: MemberEntity[];
}

export class MembersPage extends React.Component<{}, State> {
  constructor() {
    super();
    this.state = { members: [] };
  }

  public componentDidMount() {
    memberAPI.fetchMembers()
      .then((members) => {
        this.setState({ members });
      });
  }

  public render() {
    return (
      <div className="row">
        <h2> Members Page</h2>
        <table className="table">
          <thead>
            <MemberHeader />
          </thead>
          <tbody>
            {
              this.state.members.map((member) =>
                <MemberRow
                  key={member.id}
                  member={member}
                />
              )
            }
          </tbody>
        </table>
      </div>
    );
  }
};
