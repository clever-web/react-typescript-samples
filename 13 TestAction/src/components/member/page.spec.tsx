import * as React from 'react';
import {MemberPage, Props} from './page';
import * as SaveMember from './actions/saveMember';
import {saveMemberAction} from './actions/saveMember';
import {shallow} from 'enzyme';
import { createEmptyMember } from '../../reducers/member';
import { createEmptyMemberErrors } from '../../reducers/memberErrors';
import { MemberEntity } from '../../model';
import configureStore from 'redux-mock-store';
import reduxThunk from 'redux-thunk';

const middlewares = [reduxThunk];
const getMockStore = configureStore(middlewares);

describe ('/components/member/page specs',()=>{
    it('should retur as expected when definde props',()=>{
        //arrange
        const data:Props = {
            memberId:-1,
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => {},
            onChange: () =>{},
            onSave: ()=>{}
        }
        //act
        const component = shallow (
            <MemberPage {...data} />
        );
        //assert
        expect(component).toMatchSnapshot();
    });

    it('should call onSave when onSave is activated',()=>{
        //arrange
        const newMember:MemberEntity = {
            id:56789,
            avatar_url:'test new avatar',
            login: 'test new login'
        }
        const data: Props = {
            memberId: -1,
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => { },
            onChange: () => { },
            onSave: jest.fn(),
        }
        const store = getMockStore();

        const actionSaveStub = jest.spyOn(SaveMember, 'saveMemberAction')
             .mockImplementation(()=>(
               { ...newMember,}
            ));
               
        const component = shallow(
            <MemberPage {...data} />,
            {
            context: { store },
            },
        );

        
        component.prop('onSave')();
        console.log(`pete ${component}`);
        //assert
        expect(actionSaveStub).toHaveBeenCalledWith(newMember);

    });

    /* it('should call onChange when Changes are made', () => {
        //arrange

        const data: Props = {
            memberId: -1,
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => { },
            onChange: jest.fn(),
            onSave: () => { } ,
        }


        //act
        const component = shallow(
            <MemberPage {...data} />
        );


        component.find('data.onChange');
        //assert
        expect(component).toHaveBeenCalled();

    }); */
});