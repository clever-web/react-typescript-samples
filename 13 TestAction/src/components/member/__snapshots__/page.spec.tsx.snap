// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/member/page specs should retur as expected when definde props 1`] = `
<Component
  member={
    Object {
      "avatar_url": "",
      "id": -1,
      "login": "",
    }
  }
  memberErrors={
    Object {
      "login": FieldValidationResult {
        "errorMessage": "",
        "key": "",
        "succeeded": false,
        "type": "",
      },
    }
  }
  onChange={[Function]}
  onSave={[Function]}
/>
`;
